### ISSUE TITLE
Bad performance with big packages

### ISSUE THREAD

**Issue by kidandcat** [parcel-bundler/parcel#1969]
# 🐛 bug report

When importing this package using named exports https://www.npmjs.com/package/@material-ui/icons
it freezes ParcelJS.

## 🎛 Configuration (.babelrc, package.json, cli command)

Default, using jsx

## 🤔 Expected Behavior

Parcel should build as always

## 😯 Current Behavior

It freezes and consumes all CPU and Memory

## 💁 Possible Solution

Could it be possible to parse only the files that are being imported by analyzing indexes (index.js)?

## 🔦 Context

Trying to import using named exports from a big big package

## 💻 Code Sample

```javascript
import { AccessAlarm, ThreeDRotation } from '@material-ui/icons';
```

## 🌍 Your Environment

| Software         | Version(s) |
| ---------------- | ---------- |
| Parcel           | 1.9.6
| Node             | 8.10.0
| npm/Yarn         | 1.9.2
| Operating System |  Ubuntu 18.04.1 LTS

**Comment by DeMoorJasper** [comment]
It's probably better if you use something like this: `import AccessAlarm from '@material-ui/icons/AccessAlarm';` in the meantime, so it does not crash.

But I totally agree it would be amazing if parcel would be smarter in handling imports and allow adding of metadata to dependencies (Possible in Parcel 2). Sort of an explicit pre-optimisation treeshaking.

```Javascript
Dependency {
  location: string, // Absolute path in the users filesystem
  url: string, // the original string used to import the asset
  meta: object // Whatever data the transform that discovered the asset wants to pass
}

let dep = {
  location: '/Users/.../project/node_modules/@material-ui/icons/index.js',
  url: '@material-ui/icons',
  meta: {
    requiredKeys: ['AccessAlarm', 'ThreeDRotation'] // Importing everything would be `*` or a non existing `requiredKeys` in metadata
  }
}
```

For the (current) dependency typing for Parcel 2 see https://github.com/parcel-bundler/parcel/blob/next/types.js#L37 (This might and will probably change)

**Comment by alex996** [comment]
Same issue here. It bundles `@material-ui/core` fine, with or without `--experimental-scope-hoisting`. But it freezes up when using named imports on `@material-ui/icons`. I tried both with and without the flag. Same OS, Parcel `1.9.7`. Is there anything that can be done for now to allow for named imports? Thanks

**Comment by Stradivario** [comment]
@DeMoorJasper the we don't have Configuration less bundler :(

**Comment by DeMoorJasper** [comment]
@Stradivario not sure what you mean? The dependency object is just how parcel internally passes and manipulates dependencies

<sub>Sent with <a href="http://githawk.com">GitHawk</a></sub>

**Comment by devongovett** [comment]
Related: #1567

**Comment by devongovett** [comment]
Implemented in #2967

**Comment by kidandcat** [comment]
Fixed in Parcel 2
